src/client/mumble.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+Minetest
+Copyright (C) 2016 ShadowNinja <shadowninja@minetest.net>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "client/mumble.h"
+#include "util/string.h"
+#include "util/numeric.h"
+#include <cassert>
+#include <iostream>
+
+#ifdef WIN32
+	#include <windows.h>
+#else
+	#include <unistd.h>
+	#include <sys/mman.h>
+	#include <sys/stat.h>
+	#include <fcntl.h>
+#endif
+
+
+MumbleManager::~MumbleManager()
+{
+#ifdef WIN32
+	UnmapViewOfFile(lm);
+	if (obj != NULL)
+		CloseHandle(obj);
+#else
+	munmap(lm, sizeof(*lm));
+	if (fd > 0)
+		close(fd);
+#endif
+}
+
+
+bool MumbleManager::start(const std::string &context,
+		const std::string &player_name)
+{
+	assert(lm == NULL);  // Pre-condition
+#ifdef WIN32
+	obj = OpenFileMappingW(FILE_MAP_ALL_ACCESS, FALSE, L"MumbleLink");
+	if (obj == NULL)
+		return false;
+
+	lm = static_cast<MumbleLinkedMemory *>(MapViewOfFile(obj,
+			FILE_MAP_ALL_ACCESS, 0, 0, sizeof(lm)));
+	if (lm == NULL) {
+		CloseHandle(obj);
+		obj = NULL;
+		return false;
+	}
+#else
+	std::string name = "/MumbleLink." + to_string(getuid());
+
+	fd = shm_open(name.c_str(), O_RDWR, S_IRUSR | S_IWUSR);
+
+	if (fd < 0) {
+		std::cout << "Failed to create SHM for mumble!" << std::endl;
+		return false;
+	}
+
+	lm = static_cast<MumbleLinkedMemory *>(mmap(NULL, sizeof(lm),
+			PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0));
+	if (lm == MAP_FAILED) {
+		std::cout << "Failed to map SHM for mumble!" << std::endl;
+		lm = NULL;
+		return false;
+	}
+#endif
+
+	// Context is equal for players on the same server
+	// TODO: allow mods to append something to allow for teams
+	size_t len = MYMIN(sizeof(lm->context), context.size());
+	memcpy(lm->context, context.c_str(), len);
+	lm->context_len = len;
+
+	std::wstring wpn = narrow_to_wide(player_name);
```

Comment:
@ShadowNinja its stored as wstring when entered into the formspec, then its converted UTF-8 in guiformspec.cpp, and throughout the engine UTF-8 is used. Only place its converted to widestring is for chat I think because there we send wstrings over the line atm. So most probably it will be in utf8. Not that it would matter, but i'd like to remove the narrow_to_wide function once we use utf-8 based chat.